py3status/modules/ping_host.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+"""
+Display whether a hostname or IP address responds to pings.
+
+This module pings a specified hostname or IP address and displays whether the
+host is "up" or "down". The ping timeout, up text, and down text are configurable.
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module (default 2)
+    down_text: text to display if host does not respond (default '{host} is down')
+    host: hostname or IP address to ping (default "www.google.com")
+    ping_timeout: seconds to wait for ping responses (default 2)
+    up_text: text to display if host responds (default '{host} is up')
```

Comment:
This is missing at least a `format`. All modules should have `format`.

a) Good choice. To have `up_text`, `down_text` renamed to `format_up`, `format_down` since we're passing the parameters through `self.py3.safe_format()`.

b) Partial formatter choice. To have everything done in one `format`. The new `format` would look something like this... `'{host} is [\?if=is_connected up|down]'`. `is_connected` or `is_responded` or whatever you think makes sense... ought to return `True/False`.

c) Full formatter choice. This includes colors, longer string and potential confusion.
    `format = '\?if=is_connected&color=good {host} is up|\?color=bad {host} is down'`.